src/test/java/net/snowflake/ingest/streaming/internal/ParameterProviderTest.java

```diff
@@ -123,4 +131,115 @@ public void withDefaultValues() {
+  }
+
+  @Test
+  public void testMaxClientLagEnabledMinuteTimeUnitPluralTimeUnit() {
+    Properties prop = new Properties();
+    Map<String, Object> parameterMap = getStartingParameterMap();
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG_ENABLED, true);
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "2 minutes");
+    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop);
+    Assert.assertEquals(120000, parameterProvider.getBufferFlushIntervalInMs());
+  }
+
+  @Test
+  public void testMaxClientLagEnabledDefaultValue() {
+    Properties prop = new Properties();
+    Map<String, Object> parameterMap = getStartingParameterMap();
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG_ENABLED, true);
+    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop);
+    Assert.assertEquals(1000, parameterProvider.getBufferFlushIntervalInMs());
+  }
+
+  @Test
+  public void testMaxClientLagEnabledMissingUnit() {
+    Properties prop = new Properties();
+    Map<String, Object> parameterMap = getStartingParameterMap();
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG_ENABLED, true);
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "1");
+    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop);
+    try {
+      parameterProvider.getBufferFlushIntervalInMs();
+      Assert.fail("Should not have succeeded");
+    } catch (IllegalArgumentException e) {
+      Assert.assertTrue(e.getMessage().startsWith("Failed to parse"));
+    }
+  }
+
+  @Test
+  public void testMaxClientLagEnabledMissingUnitTimeUnitSupplied() {
+    Properties prop = new Properties();
+    Map<String, Object> parameterMap = getStartingParameterMap();
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG_ENABLED, true);
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, " year");
+    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop);
+    try {
+      parameterProvider.getBufferFlushIntervalInMs();
+      Assert.fail("Should not have succeeded");
+    } catch (IllegalArgumentException e) {
+      Assert.assertTrue(e.getMessage().startsWith("Failed to parse"));
+    }
+  }
+
+  @Test
+  public void testMaxClientLagEnabledInvalidTimeUnit() {
+    Properties prop = new Properties();
+    Map<String, Object> parameterMap = getStartingParameterMap();
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG_ENABLED, true);
+    parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "1 year");
+    ParameterProvider parameterProvider = new ParameterProvider(parameterMap, prop);
+    try {
+      parameterProvider.getBufferFlushIntervalInMs();
+      Assert.fail("Should not have succeeded");
+    } catch (IllegalArgumentException e) {
+      Assert.assertTrue(e.getMessage().startsWith("Invalid time unit"));
+    }
+  }
+
+  @Test
+  public void testMaxClientLagEnabledInvalidUnit() {
```

Comment:
Should we try these 2 test cases?
parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "0 seconds");
parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "");

Reply:
maybe even negative time and float times too
parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "-2 seconds");
parameterMap.put(ParameterProvider.MAX_CLIENT_LAG, "1.5 seconds");

Reply:
What should the behavior be for 0 seconds?

Reply:
NVM I set a min/max.